crates/iceberg-catalog/src/api/management/v1/warehouse.rs

```diff
@@ -529,29 +623,52 @@ pub trait Service<C: Catalog, A: AuthZHandler, S: SecretStore> {
         )
         .await?;
 
-        Ok(ListDeletedTabularsResponse {
-            tabulars: tabulars
-                .into_iter()
-                .map(|(k, (ident, delete_opts))| {
-                    let i = ident.into_inner();
-                    let deleted = delete_opts.ok_or(ErrorModel::internal(
-                        "Expected delete options to be Some, but found None",
-                        "InternalDatabaseError",
-                        None,
-                    ))?;
-
-                    Ok(DeletedTabularResponse {
-                        id: *k,
-                        name: i.name,
-                        namespace: i.namespace.inner(),
-                        typ: k.into(),
-                        warehouse_id: *warehouse_id,
-                        created_at: deleted.created_at,
-                        deleted_at: deleted.deleted_at,
-                        expiration_date: deleted.expiration_date,
-                    })
+        let tabulars = tabulars
+            .into_iter()
+            .map(|(k, (ident, delete_opts))| {
+                let i = ident.into_inner();
+                let deleted = delete_opts.ok_or(ErrorModel::internal(
+                    "Expected delete options to be Some, but found None",
+                    "InternalDatabaseError",
+                    None,
+                ))?;
+
+                Ok(DeletedTabularResponse {
+                    id: *k,
+                    name: i.name,
+                    namespace: i.namespace.inner(),
+                    typ: k.into(),
+                    warehouse_id: *warehouse_id,
+                    created_at: deleted.created_at,
+                    deleted_at: deleted.deleted_at,
+                    expiration_date: deleted.expiration_date,
                 })
-                .collect::<Result<Vec<_>>>()?,
+            })
+            .collect::<Result<Vec<_>>>()?;
+
+        let tabulars = futures::future::try_join_all(tabulars.iter().map(|t| match t.typ {
+            TabularType::View => authorizer.is_allowed_view_action(
+                &request_metadata,
+                warehouse_id,
+                t.id.into(),
+                &crate::service::authz::ViewAction::CanShowInList,
+            ),
+            TabularType::Table => authorizer.is_allowed_table_action(
+                &request_metadata,
+                warehouse_id,
+                t.id.into(),
+                &crate::service::authz::TableAction::CanShowInList,
+            ),
+        }))
+        .await?
+        .into_iter()
+        .zip(tabulars.into_iter())
+        .filter_map(|(allowed, tabular)| if allowed { Some(tabular) } else { None })
+        .collect();
+
+        // ToDo: Better pagination with non-empty pages
```

Comment:
do you see another option? Could invert it by getting all accessible tabulars from openfga & using them as a filter for listing, that could end up with a huge number of filters, does openfga support pagination?

Reply:
Following the openfga guide: https://openfga.dev/docs/interacting/search-with-permissions I think this is the best way to do it.